dashboard/app/models/user.rb

```diff
@@ -1766,7 +1766,11 @@ def self.track_level_progress(user_id:, level_id:, script_id:, new_result:, subm
       user_level.attempts += 1 unless user_level.perfect? && user_level.best_result != ActivityConstants::FREE_PLAY_RESULT
       user_level.best_result = new_result if user_level.best_result.nil? ||
         new_result > user_level.best_result
+
       user_level.submitted = submitted
+      if submitted && level_source_id
```

Comment:
Just curious: what is a `level_source_id`?

Reply:
Thank you for reminding me to double check this- it's meant to indicate whether or not a user_level is part of a level_group. I'll be updating this check based on feedback in the dashboard-dev slack channel.

Reply:
from Dave's message, looks like this maybe should be `user_level.level.type === 'LevelGroup'`. regardless of what you need to check, this would be clearer if you store the resullt in an `is_level_group` variable (e.g. `is_level_group = ...`) so that you can write this line as `if submitted && is_level_group`.